Plugins/Mute/MuteManager.cs

```diff
@@ -1,21 +1,173 @@
+            CleanedName = client.CleanedName,
+            Reason = muteState is null ? string.Empty : _translationLookup["PLUGINS_MUTE_MIGRATED"],
+            Expiration = muteState switch
+            {
+                null => DateTime.UtcNow,
+                MuteState.Muted => null,
+                _ => DateTime.UtcNow
+            },
+            AdminId = Utilities.IW4MAdminClient().ClientId,
+            MuteState = muteState ?? MuteState.Unmuted
+        };
+
+        // Migrate old mute meta, else, client has no state, so set a generic one, but don't write it to database.
+        if (muteState is not null)
+        {
+            await WritePersistentData(client, clientMuteMeta);
+            await UpdateMuteList(clientMuteMeta);
+        }
+        else
+        {
+            client.SetAdditionalProperty(Plugin.MuteKey, clientMuteMeta);
+        }
+
+        return clientMuteMeta;
+    }
+
+    public async Task<bool> Mute(Server server, EFClient origin, EFClient target, DateTime? dateTime, string reason)
     {
-        if (await Plugin.DataManager.ReadPersistentData(gameEvent.Target) == MuteState.Muted)
+        var clientMuteMeta = await GetCurrentMuteState(target);
+        if (clientMuteMeta.MuteState is MuteState.Muted && clientMuteMeta.CommandExecuted) return false;
+
+        clientMuteMeta = new MuteStateMeta
         {
-            await gameEvent.Owner.ExecuteCommandAsync($"unmute {gameEvent.Target.ClientNumber}");
-            await Plugin.DataManager.WritePersistentData(gameEvent.Target,
-                gameEvent.Target.IsIngame ? MuteState.Unmuted : MuteState.Unmuting);
-            return false;
+            ClientId = target.ClientId,
+            CleanedName = target.CleanedName,
+            Expiration = dateTime,
+            MuteState = MuteState.Muted,
+            Reason = reason,
+            AdminId = origin.ClientId,
+            CommandExecuted = false
+        };
+        await WritePersistentData(target, clientMuteMeta);
+
+        // Handle mute list
+        await UpdateMuteList(clientMuteMeta);
+
+        // Mute client
+        var client = server.GetClientsAsList().FirstOrDefault(x => x.NetworkId == target.NetworkId);
+        if (client is not null)
+        {
+            await server.ExecuteCommandAsync($"muteClient {client.ClientNumber}");
+            clientMuteMeta.CommandExecuted = true;
         }
 
-        await gameEvent.Owner.ExecuteCommandAsync($"muteClient {gameEvent.Target.ClientNumber}");
-        await Plugin.DataManager.WritePersistentData(gameEvent.Target, MuteState.Muted);
         return true;
     }
+
+    public async Task<bool> Unmute(Server server, EFClient origin, EFClient target, string reason)
+    {
+        var clientMuteMeta = await GetCurrentMuteState(target);
+        if (clientMuteMeta.MuteState is MuteState.Unmuted && clientMuteMeta.CommandExecuted) return false;
+        if (!target.IsIngame && clientMuteMeta.MuteState is MuteState.Unmuting) return false;
```

Comment:
If muting from the webfront it may execute on a server the client isn't currently on. You should use the logic on like 115 to find the ingame client first.
eg: target = server.GetClientsAsList().FirstOrDefault(x => x.NetworkId == target.NetworkId) ?? target;

Then check the .IsIngame there